### ISSUE TITLE
Installing smaller subset of the libraries

### ISSUE THREAD

**Issue by agateau** [miurahr/aqtinstall#126]
It would be great if it was possible to install a smaller subset of the Qt libraries. For example if my application does not use QML there is no need to install it, or QtQuick or QtQuickControls.

This would speed up CI builds.

**Comment by miurahr** [comment]
@agateau Could you mind to explain how to retrieve target package list form https://download.qt.io/online/qtsdkrepository/mac_x64/desktop/qt5_5150/Updates.xml 
for a smaller subset?

For example, current implementation see line 6
```
<Name>qt.qt5.5150.clang_64</Name>
```
and then get a download target package list from line 18
```
<DownloadableArchives>
qtbase-MacOS-MacOS_10_13-Clang-MacOS-MacOS_10_13-X86_64.7z, qtconnectivity-MacOS-MacOS_10_13-Clang-MacOS-MacOS_10_13-X86_64.7z, qtdeclarative-MacOS-MacOS_10_13-Clang-MacOS-MacOS_10_13-X86_64.7z, qtgraphicaleffects-MacOS-MacOS_10_13-Clang-MacOS-MacOS_10_13-X86_64.7z, qtimageformats-MacOS-MacOS_10_13-Clang-MacOS-MacOS_10_13-X86_64.7z, qtmacextras-MacOS-MacOS_10_13-Clang-MacOS-MacOS_10_13-X86_64.7z, qtmultimedia-MacOS-MacOS_10_13-Clang-MacOS-MacOS_10_13-X86_64.7z, qtsensors-MacOS-MacOS_10_13-Clang-MacOS-MacOS_10_13-X86_64.7z, qtsvg-MacOS-MacOS_10_13-Clang-MacOS-MacOS_10_13-X86_64.7z, qttranslations-MacOS-MacOS_10_13-Clang-MacOS-MacOS_10_13-X86_64.7z, qtwebchannel-MacOS-MacOS_10_13-Clang-MacOS-MacOS_10_13-X86_64.7z, qtwebsockets-MacOS-MacOS_10_13-Clang-MacOS-MacOS_10_13-X86_64.7z, qtxmlpatterns-MacOS-MacOS_10_13-Clang-MacOS-MacOS_10_13-X86_64.7z, qtlocation-MacOS-MacOS_10_13-Clang-MacOS-MacOS_10_13-X86_64.7z, qtserialport-MacOS-MacOS_10_13-Clang-MacOS-MacOS_10_13-X86_64.7z, qttools-MacOS-MacOS_10_13-Clang-MacOS-MacOS_10_13-X86_64.7z, qtquickcontrols-MacOS-MacOS_10_13-Clang-MacOS-MacOS_10_13-X86_64.7z, qtquickcontrols2-MacOS-MacOS_10_13-Clang-MacOS-MacOS_10_13-X86_64.7z, qt3d-MacOS-MacOS_10_13-Clang-MacOS-MacOS_10_13-X86_64.7z, qtwebview-MacOS-MacOS_10_13-Clang-MacOS-MacOS_10_13-X86_64.7z, qtserialbus-MacOS-MacOS_10_13-Clang-MacOS-MacOS_10_13-X86_64.7z, qtscxml-MacOS-MacOS_10_13-Clang-MacOS-MacOS_10_13-X86_64.7z, qtgamepad-MacOS-MacOS_10_13-Clang-MacOS-MacOS_10_13-X86_64.7z, qtspeech-MacOS-MacOS_10_13-Clang-MacOS-MacOS_10_13-X86_64.7z, qtremoteobjects-MacOS-MacOS_10_13-Clang-MacOS-MacOS_10_13-X86_64.7z, qt-5.15.0-macx-clang.7z
</DownloadableArchives>
```

To reduce a list, we need some logic to do it.

It seems there are no subset definition in it.

**Comment by agateau** [comment]
I think the easiest solution is to provide a way for users to specify individual archives, and install all if not set. This way it is not up to aqtinstall to do the right thing.

In my case I think I would need only qtbase, qtimageformats, qtsvg, qttranslations, qttools, and probably qt (not sure what that one contains), so if I could call aqtinstall with a `--archives` option to pass the list or archives to install, similar to the `--modules` option, it would save quite some time.

**Comment by miurahr** [comment]
v0.9.0b2 release includes `--archives` option and feature. Does it work for @agateau ?

**Comment by agateau** [comment]
Going to give it a try!

**Comment by agateau** [comment]
It works beautifully, thanks!

Installing only qtbase, qtimageformats, qtsvg, qttranslations and qttools reduced the duration of the "Qt install" step from 38s to 8s.

**Comment by pestophagous** [comment]
Thanks to this issue I learned of the existence and purpose of the `--archives` option. Thanks for adding this!

Since this issue showed up in a web search and served as good documentation for me, I thought I should add this subtle problem that I found. It has an easy solution, which I will also describe.

My environment is Ubuntu 18.

I began by running:
```
python3 -m aqt install --outputdir /tmp/myQt 5.15.0 linux desktop
```

I noticed that this was downloading `qt3d` and some other things I didn't need.

So then I discovered `--archives` and changed my command to:
```
  python3 -m aqt install --outputdir /tmp/myQt 5.15.0 linux desktop --archives \
   qtbase \
   qtconnectivity \
   qtdeclarative \
   qtquickcontrols \
   qtquickcontrols2 \
   qtsvg \
   qttools

```

However, at least on Ubuntu 18, that did not result in a fully working solution, because I saw errors like the following when trying to run `uic` and other tools:

```
error while loading shared libraries: libicui18n.so.56: cannot open shared object file: No such file or directory
```

I went back through the aqt logs and I noticed that when I invoked the NON-archives command, the downloads included `5.15.0-0-202005140804icu-linux-Rhel7.2-x64.7z`.

When I manually downloaded and unzipped `5.15.0-0-202005140804icu-linux-Rhel7.2-x64.7z` I could confirm that this contains the `libicui18n.so.56` that I was lacking when I tried the `--archives` command.

This led me to what began as a "lucky guess."  I thought that maybe I can list `icu` as an archive name. It works!

So now my command uses `--archives` and also contains `icu`, and it is working:

```
  python3 -m aqt install --outputdir /tmp/myQt 5.15.0 linux desktop --archives \
   icu \
   qtbase \
   qtconnectivity \
   qtdeclarative \
   qtquickcontrols \
   qtquickcontrols2 \
   qtsvg \
   qttools

```

Thank you for this great tool!
